### DISCUSSION TITLE
Option for custom CDN endpoint when using S3 file store

### DISCUSSION THREAD

**Discussion by gmemstr** [asciinema/discussions#301] [Ideas]
When using an S3 file store, it looks like files are fetched using a presigned URL direct from the bucket. This is *probably* okay for most people, but it would be nice to be able to define a custom CDN endpoint, for example if you want to put Cloudfront in front of the bucket with a custom URL or proxying a Cloudflare R2 bucket. This would let us stick with a single domain and configure caching to avoid excessive requests to the bucket.

I'm not familiar with Elixir, but I'm willing to give it a shot at implementing.

**Comment by ku1ik** [comment]
I actually thought about this too. Seems like a good idea.

We're serving files from a private bucket (i.e. objects themselves being private, requiring pre-signing). Do you have examples of how e.g. Cloudflare, Fastly, Cloudfront URLs would look like when compared to the original bucket URL? I assume simple domain substitution would only work for fully public objects, right?

**Comment by ku1ik** [comment]
The recent server releases has been reworked in this regard. All the assets are now served directly by the server, and the files stored on S3/R2 are first fetched and cached locally. So no excessive requests to S3, and everything goes from a single domain.
